plugins/polio/js/src/domains/LQAS-IM/LQAS/LqasAfroOverview/ListView/LqasAfroList.tsx

```diff
@@ -0,0 +1,372 @@
+    const tableClasses: Record<string, string> = useTableStyle();
+    const { bounds } = useContext(LqasAfroOverviewContext);
+    const { formatMessage } = useSafeIntl();
+    const currentView =
+        side === 'left'
+            ? router.params.displayedShapesLeft
+            : router.params.displayedShapesRight;
+
+    const showCountries = currentView !== 'district';
+    const [rowsPerPage, setRowsPerPage] = useState(10);
+    const [orderBy, setOrderBy] = useState('asc');
+    const [sortFocus, setSortFocus] = useState<SortFocus>('COUNTRY');
+    const [page, setPage] = useState<number>(0);
+    const selectedRound = getRound(router.params.rounds, side);
+
+    const { data: countryShapes, isFetching: isFetchingCountries } =
+        useAfroMapShapes({
+            category: 'lqas',
+            enabled: showCountries,
+            params: router.params as AfroMapParams,
+            selectedRound,
+            side,
+        });
+
+    const { data: districtShapes, isFetching: isFetchingDistricts } =
+        useGetZoomedInShapes({
+            bounds: JSON.stringify(bounds),
+            category: 'lqas',
+            enabled: !showCountries,
+            params: router.params as AfroMapParams,
+            selectedRound,
+            side,
+        });
+
+    const shapesToDisplay = useMemo(() => {
+        return showCountries ? countryShapes : districtShapes;
+    }, [countryShapes, districtShapes, showCountries]);
+
+    const handleSort = useCallback(
+        columnToSortBy => {
+            if (sortFocus !== columnToSortBy) {
+                setSortFocus(columnToSortBy);
+            } else if (orderBy === 'asc') {
+                setOrderBy('desc');
+            } else {
+                setOrderBy('asc');
+            }
+        },
+        [orderBy, sortFocus],
+    );
+
+    const handleChangePage = (_event, newPage) => {
+        setPage(newPage);
+    };
+
+    const handleChangeRowsPerPage = event => {
+        setRowsPerPage(parseInt(event.target.value, 10));
+        setPage(0);
+    };
+
+    const shapesForTable = useMemo(() => {
+        if (!shapesToDisplay) {
+            return null;
+        }
+        let result = cloneDeep(shapesToDisplay);
+
+        if (sortFocus === 'REGION') {
+            result = sortBy(result, ['data.region_name']);
+        } else if (sortFocus === 'COUNTRY') {
+            result = sortBy(result, ['country_name']);
+        } else if (sortFocus === 'DISTRICT') {
+            result = sortBy(result, ['data.district_name']);
+        } else if (sortFocus === 'STATUS') {
+            result = sortBy(result, ['status']);
+        } else if (sortFocus === 'ROUND') {
+            result = sortBy(result, ['data.round_number']);
+        } else if (sortFocus === 'CAMPAIGN') {
+            result = sortBy(result, ['data.campaign']);
+        } else if (sortFocus === 'LOCATION') {
+            result = sortBy(result, ['data.geo_json']);
+        }
```

Comment:
A switch is maybe cleaner for this no ?

Reply:
I don't know, I find `switch` to be needlessly verbose